great_expectations/rule_based_profiler/config/base.py

```diff
@@ -340,10 +340,8 @@ def __init__(
         self.name = name
         self.config_version = config_version
         self.rules = rules
-        if class_name is not None:
-            self.class_name = class_name
-        if module_name is not None:
-            self.module_name = module_name
+        self.class_name = class_name
+        self.module_name = module_name
```

Comment:
This may have been done for a particular reason but I really don't think we should have variable attributes like this. This, in addition to setattr, further detracts from the purpose of classes.

I've opted to add these back in but I'm happy to discuss how to strike a balance.

Reply:
@cdkini It is critical to have those as before (with the "if" statements); otherwise, the "missing" directive in the Marshmallow schema is ineffective, and we will get validation errors.  Happy to discuss.  Thank you!